app/models/skin.rb

```diff
@@ -498,6 +516,7 @@ def self.create_default
     end
     skin.official = true
     skin.save!
+    Rails.cache.increment('skins_generation/'+(skin.type.nil? ? ("site_skin") : skin.id.to_s) )
     skin
   end
   
```

Comment:
I think you should write the tests. This is not just one line of code, this pullreq is a lot of code trying to have the effect of expiring a lot of style block caches relying on the parent-child relationships of the skins, and even if it works perfectly right now, it could easily be broken by other changes to the skins in the future that didn't touch this line, and it would break the site in really confusing and hard-to-debug ways if there was no test failure to catch it. So IMO automated test coverage for this is pretty important.